python/rmm/tests/test_rmm.py

```diff
@@ -719,3 +719,25 @@ def test_dev_buf_circle_ref_dealloc():
     # deallocate `dbuf1` (which needs the MR alive), a segfault occurs.
 
     gc.collect()
+
+
+def test_custom_mr(capsys):
+    base_mr = rmm.mr.CudaMemoryResource()
+
+    def allocate_func(size):
+        print(f"Allocating {size} bytes")
+        return base_mr.allocate(size)
+
+    def deallocate_func(ptr, size):
+        print(f"Deallocating {size} bytes")
+        return base_mr.deallocate(ptr, size)
+
+    rmm.mr.set_current_device_resource(
+        rmm.mr.CallbackMemoryResource(allocate_func, deallocate_func)
+    )
+
+    dbuf = rmm.DeviceBuffer(size=256)
+    del dbuf
```

Comment:
Could we actually check the total memory allocation here instead of just looking for printed output?

Reply:
Not for any arbitrary `base_mr`. 

My thought here is that this test doesn't need to check that `base_mr` is behaving correctly, or really test what happens inside the callbacks. This test should just ensure that that the callbacks are indeed invoked as expected.

Maybe there's a better way to do that I'm missing. Modifying a global is one approach I guess?

Reply:
I think that's a reasonable expectation for the test, but in that case why even have a base mr? The test could remove the actual allocation from the callback entirely. Is calling the allocate function really testing anything other than the fact that you can run arbitrary Python from the callback?

Setting a global would work, but I'm also OK with output capturing for this purpose.

Reply:
Mainly because it serves as a useful example of how to use `CallbackMemoryResource`, although I realized that probably belongs in the docstring, so I added it there.